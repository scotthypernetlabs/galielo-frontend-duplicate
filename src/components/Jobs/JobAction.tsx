import { Fab, Icon, Tooltip } from "@material-ui/core";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { IconColor } from "../theme";
import React from "react";

interface JobActionProps {
  id: string;
  action?: any;
  onMouseUp?: any;
  toolTipText: string;
  icon: any;
  iconSize?: any;
  color: IconColor;
}

const JobAction: React.SFC<JobActionProps> = (props: JobActionProps) => {
  const { id, action, toolTipText, icon, color, iconSize, onMouseUp } = props;
  return (
    <Tooltip disableFocusListener title={toolTipText}>
      <Fab
        size="small"
        onClick={action}
        onMouseUp={onMouseUp}
        style={{ backgroundColor: color.background }}
        className="add-cursor"
      >
        <Icon key={id} style={{ color: color.main }}>
          {icon}
        </Icon>
      </Fab>
    </Tooltip>
  );
};

export default JobAction;
