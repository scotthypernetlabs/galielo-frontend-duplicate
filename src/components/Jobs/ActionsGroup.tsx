import {Box, IconButton, Tooltip} from "@material-ui/core";
import {linkBlue, red} from "../theme";
import {
  faArrowDown,
  faBox,
  faBoxOpen,
  faFileAlt,
  faInfo,
  faPause,
  faPlay,
  faTimes
} from "@fortawesome/free-solid-svg-icons";
import ArchiveOutlineIcon from "@material-ui/icons/Archive";
import UnarchiveIcon from "@material-ui/icons/Unarchive";
import React from "react";
import JobAction from "./JobAction";

export enum ActionDisplay {
  downloadResults,
  inProgress,
  paused
}

interface Props {
  display: ActionDisplay,
  jobId: string;

  // For download
  onClickDownload?: any;
  archiveJob?: any;
  isArchived?: boolean;

  // For job in progress
  pauseJob?: any;
  stopJob?: any;
  startJob?: any;
  openProcessLog?: any;
  openStdoutLog?: any;
}

const ActionsGroup: React.SFC<Props> = (props: Props) => {
  const {
    display,
    onClickDownload,
    jobId,
    isArchived,
    archiveJob,
    pauseJob,
    startJob,
    stopJob,
    openProcessLog,
    openStdoutLog
  } = props;

  return (
    <>
      {display == ActionDisplay.downloadResults ?
        ( <Box display="flex" alignItems="center">
          <Box mr={1}>
            <JobAction
              id={`${jobId}download`}
              action={onClickDownload}
              toolTipText="Download results"
              icon="arrow_downward"
              color={linkBlue}
            />
          </Box>
          {isArchived ?
            <JobAction
              id={`${jobId}unarchive`}
              toolTipText="Unarchive"
              icon="unarchive"
              onMouseUp={archiveJob}
              color={linkBlue}
            /> :
            <JobAction
              id={`${jobId}archive`}
              toolTipText="Archive"
              icon="archive"
              onMouseUp={archiveJob}
              color={linkBlue}
            />
          }
        </Box>
        ) : (
        <Box
          display="flex"
          flexWrap="nowrap"
          bgcolor="background.paper"
        >
          <Box mr={1}>
            {display == ActionDisplay.inProgress ?
              ( <JobAction
                id={`${jobId}pause`}
                action={pauseJob}
                toolTipText="Pause job"
                icon="pause"
                iconSize="sm"
                color={linkBlue}
              /> ):
              <JobAction
                id={`${jobId}start`}
                action={startJob}
                toolTipText="Start job"
                icon="plat"
                iconSize="sm"
                color={linkBlue}
              />
            }
          </Box>
          <Box mr={1}>
            <JobAction
              id={`${jobId}stop`}
              action={stopJob}
              toolTipText="Cancel job"
              icon="{faTimes}"
              color={red}
            />
          </Box>
          <Box mr={1}>
            <JobAction
              id={`${jobId}viewProcessLogs`}
              action={openProcessLog}
              toolTipText="Process logs"
              icon="info"
              color={linkBlue}
            />
          </Box>
          <Box mr={1}>
            <JobAction
              id={`${jobId}viewStdout`}
              action={openStdoutLog}
              toolTipText="Standard logs"
              icon="{faFileAlt}"
              color={linkBlue}
            />
          </Box>
        </Box>
        )
      }
    </>
  )
};

export default ActionsGroup;
