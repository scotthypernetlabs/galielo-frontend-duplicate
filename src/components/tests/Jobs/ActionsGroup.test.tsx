import {mount} from "enzyme";
import ActionsGroup, {ActionDisplay} from "../../Jobs/ActionsGroup";
import JobAction from "../../Jobs/JobAction";
import React from "react";

describe("ActionsGroup Component", () => {
  const componentProps = {
    display: ActionDisplay.downloadResults,
    jobId: "jobid",
    onClickDownload: jest.fn(),
    archiveJob: jest.fn(),
    isArchived: false,

    pauseJob: jest.fn(),
    stopJob: jest.fn(),
    startJob: jest.fn(),
    openProcessLog: jest.fn(),
    openStdoutLog: jest.fn()
  };

  // Create Downloads View
  const WrapperDownload = (props: any) => {
    return <ActionsGroup {...componentProps} />;
  };
  const actionsGroupDownload = mount(<WrapperDownload />);

  // Create Unarchive view
  componentProps.isArchived = true;
  const WrapperArchived = (props: any) => {
    return <ActionsGroup {...componentProps} />;
  };
  const actionsGroupArchived = mount(<WrapperArchived />);

  // Create In Progress View
  componentProps.display = ActionDisplay.inProgress;
  const WrapperInProgress = (props: any) => {
    return <ActionsGroup {...componentProps} />
  };
  const actionsGroupInProgress = mount(<WrapperInProgress />);

  // Create Paused View
  componentProps.display = ActionDisplay.paused;
  const WrapperPaused = (props: any) => {
    return <ActionsGroup {...componentProps} />
  };
  const actionsGroupsPaused = mount(<WrapperPaused />);

  it("Download results display should have the right elements", () => {
    expect(actionsGroupDownload.find(JobAction)).toHaveLength(2);
    expect(actionsGroupDownload.find("button")).toHaveLength(2);
  });

  it("Download results display should call the right functions", () => {
    actionsGroupDownload.find("button").at(0).simulate("click");
    expect(actionsGroupDownload.childAt(0).props().onClickDownload).toHaveBeenCalled();
    actionsGroupDownload.find("button").at(1).simulate("mouseup");
    expect(actionsGroupDownload.childAt(0).props().archiveJob).toHaveBeenCalled();
  });

  it("Download results archived view should have the right elements", () => {
    actionsGroupArchived.find("button").at(1).simulate("mouseup");
    expect(actionsGroupDownload.childAt(0).props().archiveJob).toHaveBeenCalled();
  });

  it("In progress display should have the right elements", () => {
    expect(actionsGroupInProgress.find(JobAction)).toHaveLength(4);
    expect(actionsGroupInProgress.find("button")).toHaveLength(4);
  });

  it("In progress display should call the right functions", () => {
    actionsGroupInProgress.find("button").at(0).simulate("click");
    expect(actionsGroupInProgress.childAt(0).props().pauseJob).toHaveBeenCalled();
    actionsGroupInProgress.find("button").at(1).simulate("click");
    expect(actionsGroupInProgress.childAt(0).props().stopJob).toHaveBeenCalled();
    actionsGroupInProgress.find("button").at(2).simulate("click");
    expect(actionsGroupInProgress.childAt(0).props().openProcessLog).toHaveBeenCalled();
    actionsGroupInProgress.find("button").at(3).simulate("click");
    expect(actionsGroupInProgress.childAt(0).props().openStdoutLog).toHaveBeenCalled();
  });

  it("Paused display should have the right elements", () => {
    expect(actionsGroupsPaused.find(JobAction)).toHaveLength(4);
    expect(actionsGroupsPaused.find("button")).toHaveLength(4);
  });

  it("Paused display should call the right functions", () => {
    actionsGroupsPaused.find("button").at(0).simulate("click");
    expect(actionsGroupsPaused.childAt(0).props().startJob).toHaveBeenCalled();
    actionsGroupsPaused.find("button").at(1).simulate("click");
    expect(actionsGroupsPaused.childAt(0).props().stopJob).toHaveBeenCalled();
    actionsGroupsPaused.find("button").at(2).simulate("click");
    expect(actionsGroupsPaused.childAt(0).props().openProcessLog).toHaveBeenCalled();
    actionsGroupsPaused.find("button").at(3).simulate("click");
    expect(actionsGroupsPaused.childAt(0).props().openStdoutLog).toHaveBeenCalled();
  });

  it("should match snapshot", () => {
    expect(actionsGroupDownload).toMatchSnapshot();
    expect(actionsGroupsPaused).toMatchSnapshot();
    expect(actionsGroupInProgress).toMatchSnapshot();
    expect(actionsGroupArchived).toMatchSnapshot();
  });
});
