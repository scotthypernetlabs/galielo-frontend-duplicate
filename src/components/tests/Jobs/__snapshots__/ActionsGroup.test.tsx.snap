// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionsGroup Component should match snapshot 1`] = `ReactWrapper {}`;

exports[`ActionsGroup Component should match snapshot 2`] = `ReactWrapper {}`;

exports[`ActionsGroup Component should match snapshot 3`] = `ReactWrapper {}`;

exports[`ActionsGroup Component should match snapshot 4`] = `ReactWrapper {}`;
